### DISCUSSION TITLE
Form elements

### DISCUSSION THREAD

**Discussion by mule-stand** [ariakit/ariakit#2711] [Questions]
Can I use form elements (like FormInput, FormError etc) outside of the form context, but just for controlled input fields and so on? Since these components already contain convenient logic for error handling and so on.

**Comment by diegohaz** [comment]
If you don't use `Form`, you'll have to manually pass the `store` prop to the components because that's what connects labels, fields, and errors. `Form` is also responsible for validating the fields on change and on blur, so you should call `store.validate()` manually as well.

Here's an example: https://stackblitz.com/edit/qcgrds?file=form%2Fstyle.css,form%2Findex.tsx&theme=dark

We can probably make that easier, so feel free to submit a feature request with a concrete use case.
